Fix data race in P2C load balancer's random number generator

## Description

The P2C load balancer has a data race detected by `go test -race`. The `rng` field (`*rand.Rand`) is not thread-safe, but `Select()` acquires only an `RLock`, allowing concurrent goroutines to call `p.rng.Intn()` simultaneously. Since `rand.Rand` mutates internal state on each call, this is a data race.

## Root Cause

In `internal/agent/lb/p2c.go`:
- `Select()` uses `p.mu.RLock()` (lines 57-58)
- Multiple concurrent `Select()` calls can execute `p.rng.Intn()` (lines 70-75) simultaneously
- `rand.Rand` is not safe for concurrent use

## Fix

Replace `*rand.Rand` with a `lockedRand` wrapper or use `rand/v2` with a per-call approach, or change `Select()` to use a write lock.

## CI Impact

This causes the Test job to fail on PR #39 with `-race` flag:
```
--- FAIL: TestP2CConcurrentOperations/concurrent_select_operations
    testing.go:1490: race detected during execution of test
```

## Acceptance Criteria
- [ ] `go test -race ./internal/agent/lb/` passes
- [ ] P2C load balancer remains performant under concurrent load

--- a/internal/agent/lb/p2c.go
+++ b/internal/agent/lb/p2c.go
@@ -31,6 +31,7 @@ type P2C struct {
 	endpoints []*pb.Endpoint
 	// Track active requests per endpoint
 	activeRequests map[string]*int64
+	rngMu          sync.Mutex
 	rng            *rand.Rand
 }
 
@@ -66,14 +67,14 @@ func (p *P2C) Select() *pb.Endpoint {
 		return healthy[0]
 	}
 
-	// Pick two random endpoints
+	// Pick two random endpoints (rng is not thread-safe, needs its own lock)
+	p.rngMu.Lock()
 	idx1 := p.rng.Intn(len(healthy))
 	idx2 := p.rng.Intn(len(healthy))
-
-	// Ensure we pick different endpoints
 	for idx1 == idx2 && len(healthy) > 1 {
 		idx2 = p.rng.Intn(len(healthy))
 	}
+	p.rngMu.Unlock()
 
 	ep1 := healthy[idx1]
 	ep2 := healthy[idx2]

